### PR TITLE
[GCI 2019] Added qemu.md

### PR DESCRIPTION
**Pull request by trungnt2910**
I've added qemu.md in content\guides\virtualizing in this website.
The images seem to be hosted in a different page (haiku-os.org/files/), which is not included here, so they are attached here.
[QEMU Screenshots.zip](https://github.com/haiku/website/files/4026560/QEMU.Screenshots.zip)

### CODE REVIEW

File: content/guides/virtualizing/qemu.md [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+## Part 2: Installing and running Haiku from an ISO image <a name="part_iso"></a>
+### Step 1. Creating a virtual machine
+When the .ISO image is ready, we can begin to create a new virtual machine (VM).
+
+Start QtEmu, then click on the "New Machine" icon (or go to `Machine -> New Machine`). A window should pop up, guiding you through the process of creating a new VM.
+
+![QtEmuGUINewMachine](/files/guides/virtualizing/qemu/02_QtEmu_GUI_New_Machine.png)
+
+#### 1. Machine name and operating system
+In this dialog, we need to give our machine a name, as well as specify the operating system that we are going to install. This allow QEMU optimize its machine for some popular OSes, and also to distinguish from your other VMs. As Haiku is not included in the list, we will choose "Other". Once you've finished, click on "Next".
```
Comment (humdingerb) [inline content/guides/virtualizing/qemu.md]:
```suggestion
In this dialog, we need to give our machine a name, as well as specify the operating system that we are going to install. This allows QEMU to optimize its machine for some popular OSes, and also to distinguish from your other VMs. As Haiku is not included in the list, we will choose "Other". Once you've finished, click on "Next".
```

File: content/guides/virtualizing/qemu.md [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+
+Also, the "maximum number of hotpluggable CPUs" must not be smaller than the virtual CPU count.
+
+The other tabs allow you to configure Graphics, Audio, and Network settings. If you have experience with virtualizing, you can change a few things here, but if you don't, it is safer to leave these tabs alone.
+
+When you're finished, press "Next".
+
+#### 4. Machine accelerator
+
+Then, choose your Machine Accelerator here. HAXM is enabled by default, as speeds up virtualization. However, this cause [bugs](https://gitlab.com/qtemu/gui/issues/28#note_244603038) in many machines. If HAXM is installed in your PC, you can press "Next" and move on.
```
Comment (humdingerb) [inline content/guides/virtualizing/qemu.md]:
```suggestion
Here you choose your Machine Accelerator. HAXM is enabled by default, as it speeds up virtualization. However, this causes [bugs](https://gitlab.com/qtemu/gui/issues/28#note_244603038) in many machines. If HAXM is installed on your PC, you can press "Next" and move on.
```

File: content/guides/virtualizing/qemu.md [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+
+![QtEmuNewAccelerator](/files/guides/virtualizing/qemu/06_QtEmu_new_accelerator.png)
+
+But if it is not, or if you are unsure, uncheck the HAXM option, then move to the TCG tab, and enable TCG here.
```
Comment (humdingerb) [inline content/guides/virtualizing/qemu.md]:
```suggestion
But if it is not, or if you are unsure, uncheck the HAXM option, then move to the TCG tab, and enable TCG.
```

File: content/guides/virtualizing/qemu.md [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+
+![QtEmuNewAcceleratorTCG](/files/guides/virtualizing/qemu/07_QtEmu_new_accelerator_TCG.png)
+
+#### 5. Machine memory
+The next thing to do is to set the amount of RAM allocated for your VM. The recommended amount is 3GB (as WebPositive alone can take 1GB of memory), but not more than half of your available RAM.
```
Comment (humdingerb) [inline content/guides/virtualizing/qemu.md]:
```suggestion
Next you set the amount of RAM allocated for your VM. The recommended amount is 3GB (as WebPositive alone can take 1GB of memory), but not more than half of your available RAM.
```

File: content/guides/virtualizing/qemu.md [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+
+![QtEmuNewRAM](/files/guides/virtualizing/qemu/08_QtEmu_new_RAM.png)
+
+#### 6. Machine virtual hard disk
+Finally, we need to allocate space for the virtual disk. Select "Create a new virtual hard disk, then press "Next".
```
Comment (humdingerb) [inline content/guides/virtualizing/qemu.md]:
```suggestion
Finally, we need to allocate space for the virtual disk. Select "Create a new virtual hard disk", then press "Next".
```

File: content/guides/virtualizing/qemu.md [hunk 6]
```diff
@@ -0,0 +1,156 @@
...
+8GB of disk space is recommended for Haiku. Also, for portability, you can select the raw image type or VMWare disk image type (.vmdk).
+
+![QtEmuNewDiskSize](/files/guides/virtualizing/qemu/10_QtEmu_new_disk_size.png)
+
+#### 7. Machine summary
+When you're done, your summary should look like this:
+
+![QtEmuNewSummary](/files/guides/virtualizing/qemu/11_QtEmu_new_summary.png)
+
+Press Finish to finish setup.
```
Comment (humdingerb) [inline content/guides/virtualizing/qemu.md]:
```suggestion
Press "Finish" to finish setup.
```

File: content/guides/virtualizing/qemu.md [hunk 7]
```diff
@@ -0,0 +1,156 @@
...
+
+The VM should boot from the CD, and Haiku's installer should open.
+
+Follow [this guide](https://www.haiku-os.org/get-haiku/installation-guide) if you don't know how to install Haiku.
+
+In the DriveSetup page, two volumes should appear. The drive at port 0 should be our virtual hard drive. Press `Disk -> Initialize` to initialize it. You can choose either "Intel Partition Map" or "GUID Partition Map", but the latter is better, and is supported by Haiku.
+
+![QtEmuDriveSetup](/files/guides/virtualizing/qemu/16_QtEmu_install_drivesetup.png)
+
+Once you have Initialized the disk, follow [the installation guide](https://www.haiku-os.org/get-haiku/installation-guide) to initialize the drive. After that, close the DriveSetup window to continue.
```
Comment (humdingerb) [inline content/guides/virtualizing/qemu.md]: I assume it's init drive, format partition:
```suggestion
Once you have initialized the disk, follow [the installation guide](https://www.haiku-os.org/get-haiku/installation-guide) to format the partition. After that, close the DriveSetup window to continue.
```

File: content/guides/virtualizing/qemu.md [hunk 8]
```diff
@@ -0,0 +1,156 @@
...
+![QtEmuHaikuDesktop](/files/guides/virtualizing/qemu/17_QtEmu_Haiku_desktop.png)
+
+
+To let Haiku run fullscreen, press <kbd>Ctrl</kbd>+<kbd>Alt</kbd>+<kbd>F</kbd>.
+You should also change the screen resolution, but not larger than the host's, else QEMU will produce bugs.
+
+## Part 3. Known issues <a name = "part_issues"></a>
+### 1. QEMU will not start after pressing the "Start machine" button
+This is a known issue in Windows 10, as mentioned above, it may be related to HAXM. Read more [here](https://gitlab.com/qtemu/gui/issues/28).
+### 2. Network does not start.
```
Comment (humdingerb) [inline content/guides/virtualizing/qemu.md]:
```suggestion
### 2. Network does not start
```

File: content/guides/virtualizing/qemu.md [hunk 9]
```diff
@@ -0,0 +1,156 @@
...
+Try enabling User Mode Network Connection in Machine Preferences. If that does not work, re-installing Haiku may help.
```
Comment (humdingerb) [inline content/guides/virtualizing/qemu.md]:
```suggestion
Try enabling "User Mode Network Connection" in "Machine Preferences". If that does not work, re-installing Haiku may help.
```

**Comment by trungnt2910** [comment]
I've committed your suggestions and added the screenshots.
Sorry for my mistakes, as I'm not a native speaker.

**Comment by trungnt2910** [comment]
For some unknown reason, trying a date later than 2019-12-31 will cause hugo to generate a 404 error when testing the website on my machine.

**Comment by humdingerb** [comment]
@trungnt2910: Please have a look at your new guide at the website once it's up there. If you find anything wrong, open another PR. Thanks!

**Comment by trungnt2910** [comment]
> @trungnt2910: Please have a look at your new guide at the website once it's up there. If you find anything wrong, open another PR. Thanks!

It seems like the your website host has some problems with the time header.
`date = "2020-01-06T21:49:28.000Z"` is somehow not supported, and you might need to move the year back to 2019 or earlier.

Currently, your website is now giving a 404 Error (Just like my PC did when I tested it locally on Hugo).

I think the header must be changed to `date = "2019-01-06T21:49:28.000Z"` or the host must be fixed somehow.

**Comment by trungnt2910** [comment]
> @trungnt2910: Please have a look at your new guide at the website once it's up there. If you find anything wrong, open another PR. Thanks!

Oh, I was wrong. I've set the time to my local time, which may be later than your host's time. This may be what makes Hugo produce 404 errors, because now I test it on my localhost, it works. Restarting the web page should fix now, I think.

**Comment by Begasus** [comment]
@humdingerb could you have a look there, still getting a 404 on that page ...

**Comment by trungnt2910** [comment]
> @humdingerb could you have a look there, still getting a 404 on that page ...

Rebuilding the server may work, because of the reasons I said below.

**Comment by Begasus** [comment]
@trungnt2910 that's why I poked @humdingerb  he should know how to handle it

**Comment by humdingerb** [comment]
> that's why I poked @humdingerb he should know how to handle it

Should I? I'm afraid I don't... :)
@waddlesplash, any ideas?

**Comment by trungnt2910** [comment]
@humdingerb Can you somehow restart the server? Because when you merged and hosted the sites your host machine might have not passed `date = "2020-01-06T21:49:28.000Z"` yet. It was the time I completed in my local time, but a time at the future, in GMT.
Restarting the server at a time 7 hours later than 21:49 works for my localhost.

**Comment by humdingerb** [comment]
I can't. That's why we call our admin @waddlesplash :)

**Comment by waddlesplash** [comment]
There was a commit since then, so a new build was already kicked off.

**Comment by humdingerb** [comment]
Confirmed. The page is now up for inspection.

**Pull request merged**